functions/__fzf_search_current_dir.fish

```diff
@@ -21,11 +21,17 @@ function __fzf_search_current_dir --description "Search the current directory. R
 
 
     if test $status -eq 0
-        # If this function was triggered with an empty command line and the only thing selected is a directory, then
-        # prepend ./ to the dir path. Because fish will attempt to cd implicitly if a directory name starting with a dot
-        # is provided, this allows the user to hit Enter one more time to quickly cd into the selected directory.
-        if test (count (commandline --tokenize)) = 0 && test (count $file_paths_selected) = 1 && test -d $file_paths_selected
-            set file_paths_selected ./$file_paths_selected
+        # If the user was in the middle of inputting the first token and only one path is selected,
+        # then prepend ./ to the selected path so that
+        # - if the path is an executable, the user can hit Enter one more time to execute it.
+        # - if the path is a directory, the user can hit Enter one more time to quickly cd into it (fish will
+        #   attempt to cd implicitly if a directory name starts with a dot)
+        if test (count $file_paths_selected) = 1
+            set commandline_tokens (commandline --tokenize)
+            set current_token (commandline --current-token)
```

Comment:
@kidonng please LMK if you have a better solution to command substitution resulting in an empty string. Seems fish doesn't allow you to do "(commandline --tokenize)" so I have to use variables.

Reply:
Yes, you have to use a variable (which IMO is not bad), see https://fishshell.com/docs/current/index.html#cartesian-products.